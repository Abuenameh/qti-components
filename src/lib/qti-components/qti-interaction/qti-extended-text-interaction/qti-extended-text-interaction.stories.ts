import './qti-extended-text-interaction';

import { html } from 'lit';
import { action } from '@storybook/addon-actions';
import { Meta, StoryObj } from '@storybook/web-components';
import { ifDefined } from 'lit/directives/if-defined.js';

type Story = StoryObj;

const meta: Meta = {
  component: 'qti-extended-text-interaction',

  argTypes: {
    base: { description: 'unsupported' },
    'string-identifier': { description: 'unsupported' },
    'expected-length': { control: { type: 'number' } },
    'pattern-mask': { control: { type: 'text' } },
    'placeholder-text': { control: { type: 'text' } },
    'max-strings': { description: 'unsupported' },
    'min-strings': { description: 'unsupported' },
    'expected-lines': { description: 'unsupported' },
    format: { options: ['plain', 'preformatted', 'xhtml'], description: 'unsupported' },
    'response-identifier': { control: { type: 'text' } },
    class: {
      options: ['', 'qti-height-lines-3', 'qti-height-lines-6', 'qti-height-lines-15'],
      control: { type: 'select' }
    },
    readonly: { control: { type: 'boolean' } },
    disabled: { control: { type: 'boolean' } }
  }
};
export default meta;

export const Interaction = {
  render: args =>
    html`
      <qti-extended-text-interaction
        @qti-register-interaction="${action(`qti-register-interaction`)}"
        @qti-interaction-response="${action(`qti-interaction-response`)}"
        response-identifier=${args['response-identifier']}
        .response=${args.response}
        ?disabled=${ifDefined(args.disabled)}
        ?readonly=${ifDefined(args.readonly)}
        placeholder-text=${ifDefined(args['placeholder-text'])}
        class="${args.class}"
        expected-length=${ifDefined(args['expected-length'])}
        pattern-mask=${ifDefined(args['pattern-mask'])}
        data-patternmask-message=${ifDefined(args['data-pattern-mask-message'])}
      >
      </qti-extended-text-interaction>
    `
};

export const patternMask = {
  render: Interaction.render,
  args: {
    'pattern-mask': '[A-Za-z]{3}',
    'data-pattern-mask-message': 'Alleen maar 3 letters toegestaan'
  }
};
