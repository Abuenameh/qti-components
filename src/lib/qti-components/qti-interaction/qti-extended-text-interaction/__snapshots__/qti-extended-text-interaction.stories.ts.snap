// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/qti-components/qti-interaction/qti-extended-text-interaction Default smoke-test 1`] = `
<div id="root-inner">
  <qti-extended-text-interaction response-identifier
                                 class
  >
  </qti-extended-text-interaction>
</div>
`;

exports[`lib/qti-components/qti-interaction/qti-extended-text-interaction PatternMask smoke-test 1`] = `
<div id="root-inner">
  <qti-extended-text-interaction response-identifier
                                 class
                                 pattern-mask="[A-Za-z]{3}"
                                 data-patternmask-message="Alleen maar 3 letters toegestaan"
  >
  </qti-extended-text-interaction>
</div>
`;
